lib/edition_tagger.rb

```diff
@@ -33,25 +36,37 @@ def add_mainstream_browse_tag(slug, tag)
     republish(slug)
   end
 
-  def add_mainstream_browse_tag_to_edition(edition, tag)
+  def add_mainstream_browse_tag_to_edition(edition, tag, primary)
     if archived_artefact?(edition)
       import_error(edition, "edition is part of an archived artefact")
       return false
     end
 
-    if duplicate_tag?(edition, tag)
-      import_error(edition, "Tag '#{tag}' was already present")
-      return false
-    end
-
-    associate_tag_with_edition(edition, tag)
+    associate_tag_with_edition(edition, tag, primary)
 
     logger.info("Updated #{edition.slug} version #{edition.version_number} (state: #{edition.state})")
     true
   end
 
-  def associate_tag_with_edition(edition, tag)
-    edition.browse_pages << tag
+  def associate_tag_with_edition(edition, tag, primary)
+    old_value = edition.browse_pages
+
+    new_value = edition.browse_pages.reject { |existing_tag|
+      existing_tag == tag
+    }
+    if primary
+      new_value.unshift(tag)
+    else
+      new_value << tag
+    end
+
+    if old_value == new_value
+      logger.info(
+        "Unchanged edition '#{edition.slug}' version #{edition.version_number}"
+      )
```

Comment:
I was mainly concerned about making it obvious when the script hadn't changed anything, but that's a good idea too.  Now implemented - I've used an `else` clause instead of an early return, though: https://github.com/alphagov/publisher/commit/faaf043aa167e302de9982ec441a0863ab367cf6#diff-e1918ccfb7ea58c9de088874023288daR65